socket.c

```diff
@@ -40,166 +40,179 @@ extern int debug;
 /*
  * gethostbyname() wrapper. Return 1 if OK, otherwise 0.
```

Comment:
```suggestion
 * Resolve hostname. Important: Caller is responsible for freeing addresses via freeaddrinfo()! Return 1 if OK, otherwise 0.
```
If you have a better idea for a comment that would be great also :)